packages/synapse-react-client/src/components/WikiMarkdownEditorButton/WikiMarkdownEditorButton.tsx

```diff
@@ -0,0 +1,155 @@
+import { ObjectType, WikiPageKey } from '@sage-bionetworks/synapse-types'
+import React, { useMemo, useState } from 'react'
+import {
+  CreateWikiPageInput,
+  useCreateWikiPage,
+  useGetRootWikiPageKey,
+  useGetWikiPage,
+} from '../../synapse-queries'
+import { Alert, Button, ButtonProps } from '@mui/material'
+import { defaults } from 'lodash-es'
+import WikiMarkdownEditor from '../WikiMarkdownEditor/WikiMarkdownEditor'
+
+export const ERROR_LOADING_WIKI_FAILED = 'Failed to load the wiki page: '
+export const DEFAULT_BUTTON_TEXT = 'Edit Wiki Page'
+const DEFAULT_BUTTON_PROPS: Omit<ButtonProps, 'onClick'> = {
+  children: DEFAULT_BUTTON_TEXT,
+  color: 'primary',
+  variant: 'contained',
+}
+
+export type WikiMarkdownEditorButtonProps = {
+  ownerObjectType: ObjectType
+  ownerObjectId: string
+  // if wikiPageId is undefined, will get (or create) the root WikiPage for this ownerObject
+  // otherwise, will get the WikiPage with the specified wikiPageId
+  wikiPageId?: string
+  buttonProps?: Omit<ButtonProps, 'onClick' | 'disabled'>
+}
+
+export const WikiMarkdownEditorButton: React.FunctionComponent<
+  WikiMarkdownEditorButtonProps
+> = (props: WikiMarkdownEditorButtonProps) => {
+  const {
+    ownerObjectType,
+    ownerObjectId,
+    wikiPageId: initialWikiPageId,
+  } = props
+
+  const buttonProps: Omit<ButtonProps, 'onClick' | 'disabled'> = defaults(
+    {},
+    props.buttonProps,
+    DEFAULT_BUTTON_PROPS,
+  )
+
+  const [open, setOpen] = useState<boolean>(false)
+  const [isCreatingWikiPage, setIsCreatingWikiPage] = useState<boolean>(false)
```

Comment:
For `isCreatingWikiPage` you can just use the value of `isPending` from `useCreateWikiPage`